### ISSUE TITLE
Mechanism for displaying messages when installing packages

### ISSUE THREAD

**Issue by chenghlee** [conda/conda#10118]
This ticket is asking for an alternative mechanism for conda package builders/maintainers to display messages to users prior to installations.

Some (many?) package builders/maintainers currently use pre-link scripts to display messages (EULAs, deprecation notices, additional installation instructions, etc.) to users prior to package installation.  However, the pre-link feature is not entirely appropriate for such a task; in particular, such scripts are generally not cross-platform, and more importantly, represent potential security hazards (code execution) to end-users.

An minimum implementation of this feature should:

* Define a well-known file name that `conda` and `conda-build` recognize as an UTF8-encoded text file containing the message(s) to display.
* Modify `conda-build` to copy the message file, if present in the recipe directory, into built packages
* Modify `conda` to display the contents of the message to `stdout` _before_ the package is installed (linked) into the target environment.  (`conda` must _not_ execute or permit execution of the contents of said file.)

An nice-to-have feature would be a flag in a recipe's `meta.yaml` that would allow package maintainers to require a "yes/no" response to the displayed message before installing the package.  This would allow, e.g., a maintainer to require users to accept a EULA before installing a package.  If implemented, this feature must also provide a non-interactive means to respond to such prompts (e.g., through the use of the existing `--yes` option or, better yet, through a new CLI option).

**Comment by chenghlee** [comment]
Edited description to provide basic requirements.

**Comment by marcelotrevisani** [comment]
it can be closed after #11123 gets integrated

**Comment by jakirkham** [comment]
Should we leave open for the yes/no functionality? This would be useful for messages (like a license) that require explicit agreement

**Comment by marcelotrevisani** [comment]
That was developed already with the pre_link message, isn't it?

**Comment by jakirkham** [comment]
Does the pre_link message include a user prompt? If so, how is that configured?

**Comment by marcelotrevisani** [comment]
That is not documented but it will prompt an y/n question

**Comment by marcelotrevisani** [comment]
https://github.com/conda/conda/pull/11123#issuecomment-1061522130

**Comment by marcelotrevisani** [comment]
and you can configure the recipe like that
https://github.com/conda/conda-build/pull/4203/files#diff-9864593b831524342b84838b9f93dea2114d6a58adf566d5f80e4c29c884fd6c

**Comment by marcelotrevisani** [comment]
it will always prompt the y/n question if the user specifies in the recipe that section with a file or a folder with text files in it

**Comment by jakirkham** [comment]
Thanks Marcelo! 🙏

So it always prompts for a response to the message (unless CLI option `-y`/`--yes` is given)?

**Comment by marcelotrevisani** [comment]
yup! that is correct :)

**Comment by jakirkham** [comment]
Ok agree we should close.

@chenghlee @jezdez, wdyt?

**Comment by kenodegard** [comment]
This looks complete.
